modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/suggestions/spi/BasicSuggestionsContributor.java

```diff
@@ -205,19 +206,35 @@ private Suggestion _getSuggestion(
 					ClassName className = _classNameLocalService.getClassName(
 						classNameId);
 
-					suggestionBuilder.attribute(
-						"assetURL",
-						_assetURLViewProvider.getAssetURLView(
-							assetRenderer, assetRendererFactory,
-							className.getClassName(), classPK,
-							liferayPortletRequest, liferayPortletResponse));
+					AssetRendererFactory<?> classNameAssetRendererFactory =
+						AssetRendererFactoryRegistryUtil.
+							getAssetRendererFactoryByClassName(
+								className.getClassName());
+
+					if (classNameAssetRendererFactory != null) {
```

Comment:
We could just set some variables in the if/else and then call _assetURLViewProvider.getAssetURLView() after the if/else.

Reply:
Thanks for the hint! It looks much cleaner now... let me know if the variables I named sound ok!